src/bidiMapper/domains/log/LogManager.ts

```diff
@@ -140,31 +148,39 @@ export class LogManager {
           cdpSessionId: this.#cdpTarget.cdpSessionId,
           executionContextId: params.exceptionDetails.executionContextId,
         });
+        if (realm === undefined) {
+          // Ignore exceptions not attached to any realm.
+          return;
+        }
 
-        this.#eventManager.registerPromiseEvent(
-          LogManager.#getExceptionText(params, realm).then((text) => ({
-            kind: 'success',
-            value: {
-              type: 'event',
-              method: ChromiumBidi.Log.EventNames.LogEntryAdded,
-              params: {
-                level: Log.Level.Error,
-                source: {
-                  realm: realm?.realmId ?? 'UNKNOWN',
-                  context: realm?.browsingContextId ?? 'UNKNOWN',
+        for (const browsingContext of realm.associatedBrowsingContexts) {
+          this.#eventManager.registerPromiseEvent(
+            LogManager.#getExceptionText(params, realm).then((text) => ({
+              kind: 'success',
+              value: {
+                type: 'event',
+                method: ChromiumBidi.Log.EventNames.LogEntryAdded,
+                params: {
+                  level: Log.Level.Error,
+                  source: {
+                    realm: realm.realmId,
+                    ...(realm instanceof WindowRealm
+                      ? {context: realm.browsingContext.id}
+                      : {}),
+                  },
```

Comment:
NIT: realm can have a method `source`, returning the realm and context if needed